packages/marshal/src/encodePassable.js

```diff
@@ -0,0 +1,423 @@
+// @ts-check
+import { getTag } from './helpers/passStyle-helpers.js';
+import { makeTagged } from './makeTagged.js';
+import { passStyleOf } from './passStyleOf.js';
+import { assertRecord } from './typeGuards.js';
+import {
+  nameForPassableSymbol,
+  passableSymbolForName,
+} from './helpers/symbol.js';
+import { recordParts } from './rankOrder.js';
+import { ErrorHelper } from './helpers/error.js';
+
+/** @typedef {import('./types.js').Passable} Passable */
+
+const { details: X, quote: q } = assert;
+const { is, fromEntries } = Object;
+
+export const zeroPad = (n, size) => {
+  const nStr = `${n}`;
+  assert(nStr.length <= size);
+  const str = `00000000000000000000${nStr}`;
+  const result = str.substring(str.length - size);
+  assert(result.length === size);
+  return result;
+};
+harden(zeroPad);
+
+// This is the JavaScript analog to a C union: a way to map between a float as a
+// number and the bits that represent the float as a buffer full of bytes.  Note
+// that the mutation of static state here makes this invalid Jessie code, but
+// doing it this way saves the nugatory and gratuitous allocations that would
+// happen every time you do a conversion -- and in practical terms it's safe
+// because we put the value in one side and then immediately take it out the
+// other; there is no actual state retained in the classic sense and thus no
+// re-entrancy issue.
+const asNumber = new Float64Array(1);
+const asBits = new BigUint64Array(asNumber.buffer);
+
+// JavaScript numbers are encoded by outputting the base-16
+// representation of the binary value of the underlying IEEE floating point
+// representation.  For negative values, all bits of this representation are
+// complemented prior to the base-16 conversion, while for positive values, the
+// sign bit is complemented.  This ensures both that negative values sort before
+// positive values and that negative values sort according to their negative
+// magnitude rather than their positive magnitude.  This results in an ASCII
+// encoding whose lexicographic sort order is the same as the numeric sort order
+// of the corresponding numbers.
+
+// TODO Choose the same canonical NaN encoding that cosmWasm and ewasm chose.
+const CanonicalNaNBits = 'fff8000000000000';
+
+const encodeBinary64 = n => {
+  // Normalize -0 to 0 and NaN to a canonical encoding
+  if (is(n, -0)) {
+    n = 0;
+  } else if (is(n, NaN)) {
+    return `f${CanonicalNaNBits}`;
+  }
+  asNumber[0] = n;
+  let bits = asBits[0];
+  if (n < 0) {
+    // XXX Why is the no-bitwise lint rule even a thing??
```

Comment:
`&` is a common `&&` typo that produces very confusing effects for the majority of web developers who have never needed nor likely know about bitwise operators. I find that it’s sufficient to disable no-bitwise at the top of a file that uses bitwise math at all, since it’s the kind of thing you either don’t use at all or use everywhere in a module.

Reply:
Done